File: signup.py
import mysql.connector
from mysql.connector import Error
from getpass import getpass

# Database connection
def create_connection():
    connection = None
    try:
        connection = mysql.connector.connect(
            host='localhost',
            user='your_username',
            password='',
            database='usersGrocery'
        )
        print("Connection to MySQL DB successful")
    except Error as e:
        print(f"The error '{e}' occurred")
    return connection

# Function to sign up a new user
def signup():
    name = input("Enter your name: ")
    email = input("Enter your email: ")
    password = getpass("Enter your password: ")
    confirm_password = getpass("Confirm your password: ")

    if password != confirm_password:
        print("Passwords do not match.")
        return

    connection = create_connection()
    cursor = connection.cursor()

    # Check if email already exists
    cursor.execute("SELECT * FROM users WHERE email = %s", (email,))
    if cursor.fetchone():
        print("Email already registered.")
        return

    # Insert user into database
    cursor.execute("INSERT INTO users (name, email, password) VALUES (%s, %s, %s)",
                   (name, email, password))  # Note: You should hash the password in a real application
    connection.commit()
    print("Signup successful! You can now log in.")

    cursor.close()
    connection.close()

if __name__ == '__main__':
    signup()

File: signup.php
<?php
// Database connection
$servername = "localhost";
$username = "root"; 
$password = " "; 
$dbname = "usersGrocery"; 

$conn = new mysqli($servername, $username, $password, $dbname);

// Check connection
if ($conn->connect_error) {
    die("Connection failed: " . $conn->connect_error);
}

// Check if form is submitted
if ($_SERVER["REQUEST_METHOD"] == "POST") {
    $name = $_POST['name'];
    $email = $_POST['email'];
    $password = $_POST['password'];
    $confirm_password = $_POST['confirm_password'];

    // Validate input
    if ($password !== $confirm_password) {
        die("Passwords do not match.");
    }

    // Check if email already exists
    $sql = "SELECT * FROM users WHERE email = '$email'";
    $result = $conn->query($sql);

    if ($result->num_rows > 0) {
        die("Email already registered.");
    }

    // Hash the password
    $hashed_password = password_hash($password, PASSWORD_DEFAULT);

    // Insert user into database
    $sql = "INSERT INTO users (name, email, password) VALUES ('$name', '$email', '$hashed_password')";

    if ($conn->query($sql) === TRUE) {
        echo "Signup successful! You can now log in.";
    } else {
        echo "Error: " . $sql . "<br>" . $conn->error;
    }
}

$conn->close();
?>
